blob/azureblob/azureblob.go

```diff
@@ -584,11 +584,14 @@ func (b *bucket) NewRangeReader(ctx context.Context, key string, offset, length
 	}, nil
 }
 
-func getSize(contentLength int64, contentRange string) int64 {
+func getSize(contentLength *int64, contentRange string) int64 {
+	if contentLength == nil && contentRange == "" {
```

Comment:
If contentLength is nil but contentRange is not empty, we will still hit a panic due to the pointer dereference on contentLength below. Let's just do a nil check on contentLength.
```suggestion
	if contentLength == nil {
```